renderer/index.js

```diff
@@ -432,24 +398,46 @@ function resumeTorrents () {
     .forEach((x) => startTorrentingSummary(x))
 }
 
+// Don't write state.saved to file more than once a second
+function saveStateThrottled () {
+  if (state.saveStateTimeout) return
+  state.saveStateTimeout = setTimeout(function () {
+    delete state.saveStateTimeout
+    saveState()
+  }, 1000)
+}
+
 // Write state.saved to the JSON state file
 function saveState () {
   console.log('saving state to ' + cfg.filePath)
 
   // Clean up, so that we're not saving any pending state
-  var copy = JSON.parse(JSON.stringify(state.saved))
+  var copy = Object.assign({}, state.saved)
   // Remove torrents pending addition to the list, where we haven't finished
   // reading the torrent file or file(s) to seed & don't have an infohash
-  copy.torrents = copy.torrents.filter((x) => x.infoHash)
-  copy.torrents.forEach(function (x) {
-    if (x.playStatus !== 'unplayable') delete x.playStatus
-  })
+  copy.torrents = copy.torrents
+    .filter((x) => x.infoHash)
+    .map(function (x) {
+      var torrent = {}
+      for (var key in x) {
+        if (key === 'progress' || key === 'torrentKey') {
+          continue // Don't save progress info or key for the webtorrent process
+        }
+        if (key === 'playStatus' && x.playStatus !== 'unplayable') {
+          continue // Don't save whether a torrent is playing / pending
+        }
+        torrent[key] = x[key]
+      }
+      return torrent
+    })
 
   cfg.write(copy, function (err) {
     if (err) console.error(err)
     ipcRenderer.send('savedState')
-    update()
   })
+
+  // Update right away, don't wait for the state to save
+  update()
```

Comment:
Ya we can probably remove that, but then we have to add `update()` calls after many of our calls to `saveState()`.